calculator_test.go

```diff
@@ -2,23 +2,293 @@ package calculator_test
 
 import (
 	"calculator"
+	"math"
+	"math/rand"
 	"testing"
+	"time"
 )
 
-func TestAdd(t *testing.T) {
+func closeEnough(a, b, epsilon float64) bool {
+	if math.Abs(a-b) < epsilon {
+		return true
+	}
+	return false
+}
+
+const testcount = 100
+const testcaseMaxsize = 10
+
+func TestAddRand(t *testing.T) {
+	t.Parallel()
+	seed := rand.NewSource(time.Now().UnixNano())
+	r := rand.New(seed)
+	count := 0
+	// run the test addcount times
+	for count < testcount {
+		// get a random size for the number of values
+		// to add together
+		nums := make([]float64, r.Intn(testcaseMaxsize))
+		// initialize nums slice to contain random values
+
+		arg1 := r.NormFloat64()
+		arg2 := r.NormFloat64()
+
+		want := arg1 + arg2
+		for i := range nums {
+			nums[i] = r.NormFloat64()
+			want += nums[i]
+		}
+		var testCases = []struct {
+			name string
+			arg1 float64
+			arg2 float64
+			nums []float64
+			want float64
+		}{
+			{name: "Random value data", arg1: arg1, arg2: arg2, nums: nums, want: want},
+		}
+		for _, tc := range testCases {
+			got := calculator.Add(tc.arg1, tc.arg2, tc.nums...)
+			if !closeEnough(tc.want, got, 0.0001) {
+				t.Errorf("Add (%f,%f,%v): want %f, got %f", tc.arg1, tc.arg2, tc.nums, tc.want, got)
+			}
+		}
+		count++
+	}
+}
+
+func TestAddMultSub(t *testing.T) {
+	t.Parallel()
+	tcs := []struct {
+		operation func(float64, float64, ...float64) float64
+		name      string
+		arg1      float64
+		arg2      float64
+		nums      []float64
+		want      float64
+	}{
+		// add test cases
```

Comment:
Actually, the way you've written the test cases makes this comment unnecessary, doesn't it?

Reply:
Yes, now that we have the func name in the operation field of the anonymous struct we do not need the comments about what type of testcases follow.